src/vsc_r_api.ts

```diff
@@ -239,7 +239,7 @@ export async function launchAddinPicker() {
   if (!config().get<boolean>('sessionWatcher')) {
     throw ('{rstudioapi} emulation requires session watcher to be enabled in extension config.');
   }
-  if (typeof rTerm === 'undefined') {
+  if (typeof sessionDir === 'undefined') {
```

Comment:
Maybe checking if `sessionDir` exists in filesystem is more robust? If self-managed terminal is killed, then the `sessionDir` still is defined but no longer exists.

Reply:
So if a self-managed terminal is killed, the `deleteTerminal` handler is run, which runs `removeSessionFiles`?

Inside `removeSessionFiles` I added a line to unset the `sessionDir`, so I think that should be equivalent?

Unless the handler isn't run. If the self-managed session can terminate without notifying VSCode to call the handler, I can't see a way to know for sure a session is attached and maybe the design needs to be changed to support self-managed sessions.

Reply:
By "a self-managed terminal is killed", I mean an R session started manually in a vscode terminal (from a vscode terminal directly, or a tmux window in the vscode terminal). In either case, `deleteTerminal` won't be called since the vscode terminal is not closed but the R session is killed.

Reply:
In other words, `deleteTerminal` is only called when user closes the VSCode terminal, rather than the R session in the terminal. If the R session is self-managed, then terminating the R session will make the VSCode terminal go back into e.g. `bash`. The R session is gone but the VSCode terminal is still there and `deleteTerminal` won't be called in this case.

Reply:
If the user-managed R session quits on R error or `q()`, then the session `tempdir()` which contains `sessionDir` will be removed. In this case, `sessionDir` is still defined but the directory it points to no longer exists.

Checking if `sessionDir` exists in filesystem will not work if the user-managed R session is killed or crashes on C level, then the `tempdir()` won't be removed.

Reply:
Okay thanks for the explanation! I have changed the test to look for the session directory. It seems that's about the best we can do easily? 

The consequences do not seem serious anyway, since if you have had a session terminate at the C level, and then run the addin picker without reattaching a new session it will probably still work? Since it will call chooseTerminal which will not find an R terminal and will create one to run the addin. So worst case I think is that you are looking at an out of date addin list?